src/workspaces/common/WorkspaceMenu.ts

```diff
@@ -20,7 +20,11 @@ import {
 } from 'src/workspaces/utils';
 
 const isNameType = (o: WorkspaceInfo): o is DynamicWorkspaceInfo =>
-  'name' in o && typeof o.name === 'string' && 'namespace' in o && typeof o.namespace === 'string';
+  'name' in o &&
+  typeof o.name === 'string' &&
+  'namespace' in o &&
+  typeof o.namespace === 'string' &&
+  Object.keys(o).length === 2;
```

Comment:
Since `LoadedWorkspaceInfo` also contains namespace and name, this check was returning true for both of them (since late 2023).